core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,153 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
+    """Decorator to check whether the user can review voiceover applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to review voiceover application.
+    """
+
+    def test_can_review_voiceover_application(
+            self, voiceover_application_id, **kwargs):
+        """Checks if the user can review voiceover application.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+            UnauthorizedUserException: The user does not have
+                credentials to voiceover an exploration.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if (
+                role_services.ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION not in
+                user_actions_info.actions):
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to review voiceover '
+                'applications.')
+
+        return handler(self, voiceover_application_id, **kwargs)
+
+    test_can_review_voiceover_application.__wrapped__ = True
+
+    return test_can_review_voiceover_application
+
+
+def can_submit_voiceover_application(handler):
+    """Decorator to check whether a user can submit voiceover application.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks
+            if the user has permission to submit voiceover application.
+    """
+
+    def test_can_submit_voiceover_application(self, **kwargs):
+        """Checks if the user can submit voiceover application.
+
+        Args:
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            UnauthorizedUserException: The user does not have
+                credentials to submit voiceover application.
+        """
+        if role_services.ACTION_SUBMIT_VOICEOVER_APPLICATION in (
+                self.user.actions):
+            # The voiceover_service returns user submitted voiceover
+            # applications in sorted order of the date of submission.
+            voiceover_applications = (
+                voiceover_services.get_user_submitted_voiceover_applications(
+                    self.user_id))
+            if voiceover_applications:
+                latest_voiceover_application = voiceover_applications[0]
+                if latest_voiceover_application.status == (
+                        suggestion_models.STATUS_IN_REVIEW):
+                    raise self.UnauthorizedUserException(
+                        'You can only submit one application at a time. '
```

Comment:
Isn't this kinda bad? What if the user wants to submit applications to different explorations?

Reply:
We are allowing one application at a time (Like we do it on Github for new contributors.)

Reply:
Ok, so I discussed this with Sean, and here is what we think would be ideal.

If a contributor is submitting their first application, this seems good (restricting to one at a time). It also sounds good that they can do only one exp at a time. However, there should be someway to set that once the voiceover for the exp is done (in its current state), the task should be marked as "complete", and they should be able to apply for another exp. If the first exp then changes, it shouldn't block the voiceover artist from applying to new ones. 

@seanlip I think this was the crux of what we talked about, did I miss anything? Also, @DubeySandeep does this sound good?